core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -231,22 +231,60 @@ typedef uint16_t l1_idx_t;
 #endif
 #endif
 
-typedef uint64_t base_xlat_tbls_t[CFG_TEE_CORE_NB_CORE][NUM_BASE_LEVEL_ENTRIES];
-typedef uint64_t xlat_tbl_t[XLAT_TABLE_ENTRIES];
+/*
+ * The global base translation table is a three dimensional array (array of
+ * array of array), but it's easier to visualize if broken down into
+ * components.
+ *
+ * TTBR is assigned a base translation table of NUM_BASE_LEVEL_ENTRIES
+ * entries.  NUM_BASE_LEVEL_ENTRIES is determined based on
+ * CFG_LPAE_ADDR_SPACE_BITS.  CFG_LPAE_ADDR_SPACE_BITS is by default 32
+ * which results in NUM_BASE_LEVEL_ENTRIES defined to 4 where one entry is
+ * a uint64_t, 8 bytes.
+ *
+ * If CFG_CORE_UNMAP_CORE_AT_EL0=y there are two base translation tables,
+ * one for OP-TEE core with full mapping of both EL1 and EL0, and one for
+ * EL0 where EL1 is unmapped except for a minimal trampoline needed to
+ * restore EL1 mappings on exception from EL0.
+ *
+ * Each CPU core is assign a unique set of base translation tables as:
+ * core0: table0: entry0 (maps both EL1 and EL0)
+ *                entry1
+ *                entry2
+ *                entry3
+ * core0: table1: entry (maps only EL0)
```

Comment:
Do you mean `entry0` here? Or is it on purpose to differentiate from `entry0` while `entry1` etc. are the same in both tables?

Reply:
It's a type, it should be `entry0`.  I'll update the comment.